### DISCUSSION TITLE
order by null value last

### DISCUSSION THREAD

**Discussion by Jeffr-K** [sequelize/sequelize#15492] [Help & Questions] [answered]
how do I place null value at the end of table using order by?

```
let themes = await GiftTheme.scope("list").findAll({
    where: {
      ...where,
    },
    order: [
      sequelize.fn('isnull', sequelize.col('categoryName')),
      ['categoryName', 'ASC'] // if categoryName is null, it must be at the end of table, print out error
    ],
  });
```

```
order: ['categoryName', 'ASC NULLS LAST'] // doesn't work for me.
```

**Comment by ephys** [comment, marked as answer]
Try this:

```ts
order: [
  // nested array, otherwise it will try to sort by a column named "ASC NULLS LAST"
  ['categoryName', 'ASC NULLS LAST']
]
```

These are the possible sort orders:

```ts
ASC
DESC
ASC NULLS LAST
DESC NULLS LAST
ASC NULLS FIRST
DESC NULLS FIRST
```

**Reply by Jeffr-K** [reply]
Thank you for reply. But my console prints out `InternalServerError: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'NULLS LAST LIMIT 0, 50' at line 1` 
What is this ? I am using mysql 8.0+ on aws cloud.

**Reply by ephys** [reply]
Can you post the complete query? You can access it by adding `logging: true` in the options of the method you're calling

**Reply by Jeffr-K** [reply]
@ephys You are my hero, although I didn't use our solution, But I resolved my issue by using ordering query like `[{ model: GiftThemeCategory, as: 'category' }, 'sort', 'ASC']`. And I added `required: true` to my model.
My interests was how can I get newest created data except null values when null values have first index. (I am sorry my english skill).
